### DISCUSSION TITLE
Libre 2 and 3 / Extend "calibrate based on glucose" to patched app data source

### DISCUSSION THREAD

**Discussion by psonnera** [NightscoutFoundation/xDrip#2878] [Ideas]
Some external apps sending Libre readings to xDrip+ don't allow calibration. xDrip+ allows a fixed slope limited intercept change of +20/-40 mg/dl that can help sometimes but doesn't cover the whole glucose range. Since Libre 2 EU OOP2 allows "calibrate based on glucose" that is applying the xDrip+ algorithm to something that is not raw data, would it make sense to extend it to external apps? This would be useful for non-EU Libre 2 and Libre 3. Safety has to be taken into consideration obviously as people usually enable plugins to achieve this (and that is NOT safe IMHO).
I'd like @keencave and @tzachi-dar comments on this if it makes sense.

**Comment by enpa23** [comment]
This would be very useful. Sometimes sensors have an offset which is consistent during days and weeks, this could be addressed by a calibration. IMO as long as the intercept coefficient is limited ( eg 0.9-1.2 ) risks are similar to the current situation where you may apply an intercept offset given by an high range measurement and this is extended to even the lowest values.
A way to add a safety layer could be to request at least one calibration in the lower range ( say 110 ) before applying it.
Also, intercept and slope coefficient safe limits could be _calibrated_  based on sensors maximum error specifications ( also with a message alert like when the difference is beyond that ), so that there is no risk of trying to calibrate a sensor which is actually faulty and has just to be replaced ( IDK if  this is how is already done ) .
